### DISCUSSION TITLE
Acct-Unique-Session-Id is changed after rekey

### DISCUSSION THREAD

**Discussion by fflrichard** [strongswan/strongswan#1512] [Q&A]
Hi, I find that the Acct-Unique-Session-Id is changed after rekey , then freeradius will insert new data. so you will see two acct data after rekey and the last data will not be handled even if the connection is closed. 
any help?

**Comment by tobiasbrunner** [comment]
I think you are confusing [reauthentication with rekeying](https://docs.strongswan.org/docs/5.9/config/rekeying.html#_ike_sas). The session ID will change during a reauthentication, it will not change during rekeyings.

**Reply by fflrichard** [reply]
> I think you are confusing [reauthentication with rekeying](https://docs.strongswan.org/docs/5.9/config/rekeying.html#_ike_sas). The session ID will change during a reauthentication, it will not change during rekeyings.

the **Acct-Session-Id** is same, but **Acct-Unique-Session-Id** is changed

here the eap-radius accounting example
```
Thu Nov 11 19:54:01 2021
        Acct-Status-Type = Start
        Acct-Session-Id = "1634624945-573"
        NAS-Port-Type = Virtual
        Service-Type = Framed-User
        NAS-Port = 573
        NAS-Port-Id = "rw"
        NAS-IP-Address = 10.10.0.150
        Called-Station-Id = "10.10.0.150[4500]"
        Calling-Station-Id = "90.166.19.115[62472]"
        User-Name = "koala.strongsec.com"
        Framed-IP-Address = 10.10.1.65
        NAS-Identifier = "strongSwan"
        Event-Timestamp = "Nov 11 2021 19:54:01 CET"
        Tmp-String-9 = "ai:"
        Acct-Unique-Session-Id = "fa1f75166d50a9845e97ad0b86d01f52"
        Timestamp = 1636656841
```

**Reply by tobiasbrunner** [reply]
Oh, I missed the significance of the `Unique` part. `Acct-Unique-Session-Id` is not actually a standardized RADIUS attribute and strongSwan does not send it. It's probably added internally by the RADIUS server, e.g. by FreeRADIUS's [`rlm_acct_unique` module](https://freeradius.org/radiusd/man/rlm_acct_unique.html). Make sure the value is generated in a way that avoids issues (in FreeRADIUS the attributes that are hashed for the value can be configured).

**Reply by fflrichard** [reply]
> Oh, I missed the significance of the `Unique` part. `Acct-Unique-Session-Id` is not actually a standardized RADIUS attribute and strongSwan does not send it. It's probably added internally by the RADIUS server, e.g. by FreeRADIUS's [`rlm_acct_unique` module](https://freeradius.org/radiusd/man/rlm_acct_unique.html). Make sure the value is generated in a way that avoids issues (in FreeRADIUS the attributes that are hashed for the value can be configured).

OK，I solved the problem now. thanks a lot. Here are some notes:

[`rlm_acct_unique` module](https://freeradius.org/radiusd/man/rlm_acct_unique.html) this module is deprecated [[Deleted Modules](https://github.com/FreeRADIUS/freeradius-server/blob/v3.0.x/raddb/README.rst#deleted-modules)].

the `Acct-Session-Id` is the same, but freeradius generate the `Acct-Unique-Session-Id` with the [`NAS-Port`](https://docs.strongswan.org/docs/6.0/plugins/eap-radius.html#_attributes_sent_to_radius_servers)  attribute by default ([accounting](https://github.com/FreeRADIUS/freeradius-server/blob/v3.0.x/raddb/policy.d/accounting)) and the `NAS-Port` is changed after rekeyed [[Add RADIUS Accounting Acct-Session-Id Attribute to Access-Request Messages](https://wiki.strongswan.org/issues/2853) ]

**Reply by fflrichard** [reply]
> Oh, I missed the significance of the `Unique` part. `Acct-Unique-Session-Id` is not actually a standardized RADIUS attribute and strongSwan does not send it. It's probably added internally by the RADIUS server, e.g. by FreeRADIUS's [`rlm_acct_unique` module](https://freeradius.org/radiusd/man/rlm_acct_unique.html). Make sure the value is generated in a way that avoids issues (in FreeRADIUS the attributes that are hashed for the value can be configured).

and here's another question: is the [`Acct-Session-Id`](https://datatracker.ietf.org/doc/html/rfc2866#section-5.5) always unique?

**Reply by tobiasbrunner** [reply]
The session IDs are composed of the timestamp when the daemon is started and the (initial, in a chain of rekeyings) IKE_SA's unique ID (that counter is reset when the daemon is restarted). So they should be reasonable unique.

**Reply by fflrichard** [reply]
> The session IDs are composed of the timestamp when the daemon is started and the (initial, in a chain of rekeyings) IKE_SA's unique ID (that counter is reset when the daemon is restarted). So they should be reasonable unique.

OK，👍🏻
